gcloud/connection.py

```diff
@@ -94,6 +93,25 @@ def http(self):
                 self._http = self._credentials.authorize(self._http)
         return self._http
 
+    @staticmethod
+    def _scoped_credentials(credentials, scope):
+        """Create a scoped set of credentials if it is required.
+
+        :type credentials: :class:`oauth2client.client.OAuth2Credentials` or
+                           :class:`NoneType`
+        :param credentials: The OAuth2 Credentials to add a scope to.
+
+        :type scope: list of URLs
+        :param scope: the effective service auth scopes for the connection.
+
+        :rtype: :class:`oauth2client.client.OAuth2Credentials` or
+                :class:`NoneType`
+        :returns: A new credentials object that has a scope added (if needed).
+        """
+        if credentials and credentials.create_scoped_required():
```

Comment:
This is a private helper method, FBO the concrete connection class constructors.  We should just trust that the pass a valid set of scopes, rather than sniffing.